cmd/tink-server/main.go

```diff
@@ -22,89 +27,246 @@ var (
 	logger log.Logger
 )
 
-func main() {
-	log, err := log.Init("github.com/tinkerbell/tink")
-	if err != nil {
-		panic(err)
+// DaemonConfig represents all the values you can configure as part of the tink-server.
+// You can change the configuration via environment variable, or file, or command flags.
+type DaemonConfig struct {
+	Facility              string
+	PGDatabase            string
+	PGUSer                string
+	PGPassword            string
+	PGSSLMode             string
+	OnlyMigration         bool
+	GRPCAuthority         string
+	TLSCert               string
+	CertDir               string
+	HTTPAuthority         string
+	HTTPBasicAuthUsername string
+	HTTPBasicAuthPassword string
+}
+
+func (c *DaemonConfig) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&c.Facility, "facility", "deprecated", "This is temporary. It will be removed")
+	fs.StringVar(&c.PGDatabase, "postgres-database", "tinkerbell", "The Postgres database name")
+	fs.StringVar(&c.PGUSer, "postgres-user", "tinkerbell", "The Postgres database username")
+	fs.StringVar(&c.PGPassword, "postgres-password", "tinkerbell", "The Postgres database password")
+	fs.StringVar(&c.PGSSLMode, "postgres-sslmode", "disabled", "Enable or disable SSL mode in postgres")
+	fs.BoolVar(&c.OnlyMigration, "only-migration", false, "When enabled it ")
+	fs.StringVar(&c.GRPCAuthority, "grpc-authority", ":42113", "The address used to expose the gRPC server")
+	fs.StringVar(&c.TLSCert, "tls-cert", "", "")
+	fs.StringVar(&c.CertDir, "cert-dir", "", "")
+	fs.StringVar(&c.HTTPAuthority, "http-authority", ":42114", "The address used to expose the HTTP server")
+}
+
+func (c *DaemonConfig) PopulateFromLegacyEnvVar() {
+	if f := os.Getenv("FACILITY"); f != "" {
+		c.Facility = f
 	}
-	logger = log
-	defer logger.Close()
-	log.Info("starting version " + version)
-
-	ctx, closer := context.WithCancel(context.Background())
-	errCh := make(chan error, 2)
-	facility := os.Getenv("FACILITY")
-
-	// TODO(gianarb): I moved this up because we need to be sure that both
-	// connection, the one used for the resources and the one used for
-	// listening to events and notification are coming in the same way.
-	// BUT we should be using the right flags
-	connInfo := fmt.Sprintf("dbname=%s user=%s password=%s sslmode=%s",
-		os.Getenv("PGDATABASE"),
-		os.Getenv("PGUSER"),
-		os.Getenv("PGPASSWORD"),
-		os.Getenv("PGSSLMODE"),
-	)
-
-	dbCon, err := sql.Open("postgres", connInfo)
-	if err != nil {
-		logger.Error(err)
-		panic(err)
+	if pgdb := os.Getenv("PGDATABASE"); pgdb != "" {
+		c.PGDatabase = pgdb
 	}
-	tinkDB := db.Connect(dbCon, logger)
-
-	_, onlyMigration := os.LookupEnv("ONLY_MIGRATION")
-	if onlyMigration {
-		logger.Info("Applying migrations. This process will end when migrations will take place.")
-		numAppliedMigrations, err := tinkDB.Migrate()
-		if err != nil {
-			log.Fatal(err)
-			panic(err)
+	if pguser := os.Getenv("PGUSER"); pguser != "" {
```

Comment:
Any reason we aren't using automatic environment variable handling?
https://github.com/tinkerbell/tink/blob/0b477fd65ffc64c1143c44c3799d2c1bc39d7437/cmd/tink-worker/cmd/root.go#L115-L121

Reply:
I see.. https://github.com/tinkerbell/tink/pull/431/files#diff-d33013e31f0cb0caf39aaef7185eec15145f19bdf8da9e32b63b37d669572f3bR128-R133